cmake/FindLpeg.cmake

```
@@ -1,4 +1,13 @@
-find_library2(LPEG_LIBRARY NAMES lpeg_a lpeg liblpeg_a lpeg${CMAKE_SHARED_LIBRARY_SUFFIX} PATH_SUFFIXES lua/5.1)
+find_library2(LPEG_LIBRARY NAMES lpeg_a lpeg liblpeg_a lpeg.so lpeg${CMAKE_SHARED_LIBRARY_SUFFIX} PATH_SUFFIXES lua/5.1)
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND LPEG_LIBRARY MATCHES ".so$")
+  execute_process(
+    COMMAND otool -hv "${LPEG_LIBRARY}"
+    OUTPUT_VARIABLE LPEG_HEADER
+    )
+  if(LPEG_HEADER MATCHES ".*BUNDLE.*")
+    message(FATAL_ERROR "lpeg library found at ${LPEG_LIBRARY} but built as a bundle rather than a dylib, please rebuild with `-shared -fPIC` rather than `-bundle`")
```

```suggestion
    message(FATAL_ERROR "lpeg library found at ${LPEG_LIBRARY} but built as a bundle rather than a dylib, please rebuild with `-dynamiclib` rather than `-bundle`")
```
This is slightly more idiomatic on macOS. (`-fPIC` is a no-op on modern macOS -- all code is position-independent.)

---

Fair point, I'll take both of these.